src/tests/java/htsjdk/samtools/sra/SRATest.java

```diff
@@ -0,0 +1,551 @@
+
+        final SAMRecordIterator samRecordIterator = reader.iterator();
+
+        int countAlignments = 0, countUnalignments = 0;
+        while (true) {
+            boolean hasRecord = samRecordIterator.hasNext();
+            SAMRecord record = null;
+            try {
+                record = samRecordIterator.next();
+                Assert.assertTrue(hasRecord); // exception is not thrown if we came to this point
+            } catch (NoSuchElementException e) {
+                Assert.assertFalse(hasRecord);
+            }
+
+            Assert.assertEquals(hasRecord, record != null);
+
+            if (record == null) {
+                break;
+            }
+
+            if (record.getReadUnmappedFlag()) {
+                countUnalignments++;
+            } else {
+                countAlignments++;
+            }
+        }
+
+        Assert.assertEquals(numberAlignments, countAlignments);
+        Assert.assertEquals(numberUnalignments, countUnalignments);
+    }
+
+    @DataProvider(name = "testCountsBySpan")
+    public Object[][] createDataForCountsBySpan() {
+        return new Object[][] {
+            {"SRR2096940", Arrays.asList(new Chunk(0, 59128983), new Chunk(59128983, 59141089)), 10591, 498},
+            {"SRR2096940", Arrays.asList(new Chunk(0, 29128983), new Chunk(29128983, 59141089)), 10591, 498},
+            {"SRR2096940", Arrays.asList(new Chunk(0, 59134983), new Chunk(59134983, 59141089)), 10591, 498},
+            {"SRR2096940", Arrays.asList(new Chunk(0, 59130000)),                                10591, 0},
+            {"SRR2096940", Arrays.asList(new Chunk(0, 59140889)),                                10591, 298}
+        };
+    }
+
+    @Test(dataProvider = "testCountsBySpan")
+    public void testCountsBySpan(String acc, List<Chunk> chunks, int numberAlignments, int numberUnalignments) {
+        if (!SRAAccession.isSupported()) return;
+
+        SamReader reader = SamReaderFactory.make().validationStringency(ValidationStringency.SILENT).open(
+                SamInputResource.of(new SRAAccession(acc))
+        );
+
+        final SAMRecordIterator samRecordIterator = ((SamReader.Indexing) reader).iterator(new BAMFileSpan(chunks));
+
+        int countAlignments = 0, countUnalignments = 0;
+        while (true) {
+            boolean hasRecord = samRecordIterator.hasNext();
+            SAMRecord record = null;
+            try {
+                record = samRecordIterator.next();
+                Assert.assertTrue(hasRecord); // exception is not thrown if we came to this point
+            } catch (NoSuchElementException e) {
+                Assert.assertFalse(hasRecord);
+            }
+
+            Assert.assertEquals(hasRecord, record != null);
+
+            if (record == null) {
+                break;
+            }
+
+            if (record.getReadUnmappedFlag()) {
+                countUnalignments++;
+            } else {
+                countAlignments++;
+            }
+        }
```

Comment:
This loop is repeated several times in this test class -- extract it into a shared helper method that returns a `Pair<alignedCount, unalignedCount>`